### ISSUE TITLE
$attr and $listeners is readonly in Rails app with turbolinks and vue/dist/vue.esm

### ISSUE THREAD

**Issue by mickenorlen** [LinusBorg/portal-vue#201]
Hi,

I'm using rails 5 and have a setup like this 
```
import TurbolinksAdapter from 'vue-turbolinks'
import Vue from 'vue/dist/vue.esm'
import PortalVue from 'portal-vue/dist/portal-vue.esm';
# Same issue with:
# import PortalVue from 'portal-vue';

Vue.use(TurbolinksAdapter)
Vue.use(PortalVue)
```

I can get the portal to render in the portal target, but when i set `<portal v-bind:to="portalTarget">` and change it dynamically. I get the errors:

```
[Vue warn]: $listeners is readonly.
[Vue warn]: $attrs is readonly.
```
From `vue.runtime.esm.js:640`

I think this isn't specifically related to vue-portal. It appears to be the `Vue.use(xxx)` that causes issues (maybe it somewhere imports 'vue' instead of 'vue.esm', causing duplicate loading?), it happened with some other plugin as well. But when i use `Vue.component('xxx')` I haven't had any problems. I couldn't find any answers so I thought I'd ask here.

Thanks for you time!

**Comment by LinusBorg** [comment]
> (maybe it somewhere imports 'vue' instead of 'vue.esm', causing duplicate loading?)

Yes, portal-vue does `import Vue from 'vue'`, and this may result in a different import than what you do, so you end up with two instances of Vue.

This is why it's strongly recommended to always alias your import, so you can also do `ìmport Vue from 'vue'`, esuring your app and any 4rd-psarty libs all import the same file. You can find examples for various bundlers in the docs here:

https://vuejs.org/v2/guide/installation.html#Runtime-Compiler-vs-Runtime-only

**Comment by mickenorlen** [comment]
Thank you so much that solved it!

In my case for rails i followed this: https://github.com/rails/webpacker/issues/1482

Created `webpack/custom-config.js`:
```
module.exports = {
	resolve: {
		alias: {
			vue: 'vue/dist/vue.js'
		}
	}
}
```
And added this to `webpack/environment.js`:
`const customConfig = require('./custom-config')`
// before module export.
`environment.config.merge(customConfig)`

Now i can use `import Vue from 'vue';` and no more warnings.

**Comment by emanuelgsouza** [comment]
I don't know if someone will stop here for any reason, but, I just have to share this other way to fix this problem in a vue-cli (v4.5.9) project.

I am using the portal-vue (**excellent package**) and, when I tried to use my lib with `yarn link` the `$attrs` and `$listeners` erros appeared on console.

I tried to use the following alias config from documentation and didn't work:

```js
module.exports = {
  // ...
  resolve: {
    alias: {
      'vue$': 'vue/dist/vue.esm.js' // 'vue/dist/vue.common.js' for webpack 1
    }
  }
}
```

I've just updated the alias to use `path.resolve` function from [this comment](https://github.com/vuetifyjs/vuetify/discussions/4068#discussioncomment-24984) and worked! So, my code now is like that:
```js
const path = require('path')

module.exports = {
  configureWebpack: {
    resolve: {
      alias: {
        vue$: path.resolve(__dirname, 'node_modules/vue/dist/vue.esm.js'),
      },
      symlinks: false, // because the related errors on try to use the `yarn link` in a vue-cli project 
    },
  },
}
```

So, thank you again for this project.

**Comment by rodrigorodriguescosta** [comment]
> I don't know if someone will stop here for any reason, but, I just have to share this other way to fix this problem in a vue-cli (v4.5.9) project.
> 
> I am using the portal-vue (**excellent package**) and, when I tried to use my lib with `yarn link` the `$attrs` and `$listeners` erros appeared on console.
> 
> I tried to use the following alias config from documentation and didn't work:
> 
> ```js
> module.exports = {
>   // ...
>   resolve: {
>     alias: {
>       'vue$': 'vue/dist/vue.esm.js' // 'vue/dist/vue.common.js' for webpack 1
>     }
>   }
> }
> ```
> 
> I've just updated the alias to use `path.resolve` function from [this comment](https://github.com/vuetifyjs/vuetify/discussions/4068#discussioncomment-24984) and worked! So, my code now is like that:
> 
> ```js
> const path = require('path')
> 
> module.exports = {
>   configureWebpack: {
>     resolve: {
>       alias: {
>         vue$: path.resolve(__dirname, 'node_modules/vue/dist/vue.esm.js'),
>       },
>       symlinks: false, // because the related errors on try to use the `yarn link` in a vue-cli project 
>     },
>   },
> }
> ```
> 
> So, thank you again for this project.

thank yoy so much!

**Comment by KennanChan** [comment]
I am creating a component library that functions as a Vue plugin without directly importing Vue. However, one of the components in the library imports portal-vue, which leads to a duplicate copy of Vue being included in the library package. This could potentially cause issues because the Vue instance used by the application that uses the component library will be a separate copy. Are there any recommended solutions for resolving this scenario?
